### PR TITLE
Removing uav simulator, renaming topics and pkgs in planner

### PR DESCRIPTION
**Pull request by VaidehiSom**
Associated [PR](https://github.com/robotics-88/vehicle-launch/pull/37)

### Setup
No special setup required

### Running
For the simulator, run `roslaunch vehicle_launch decco.launch simulate:=true slam_type:=0 planner_type:=2`
The `planner_type` arg has value `1` for our original `planner` and `2` for `ego-planner-swarm`

### CODE REVIEW

**Comment by rien88** [comment]
You'll need to merge main in on `vehicle_launch` now that Gus's PR went through, but even when I made the local changes, I'm still not getting any reaction to the flights. I see these messages in the terminal:
```
[ INFO] [1720197665.003213117]: New goal received, planning path...
[ INFO] [1720197665.004753003]: Path planning successful.
[ WARN] [1720197665.004944464]: Received empty path
```

**Comment by VaidehiSom** [comment]
> You'll need to merge main in on `vehicle_launch` now that Gus's PR went through, but even when I made the local changes, I'm still not getting any reaction to the flights. I see these messages in the terminal:
> 
> ```
> [ INFO] [1720197665.003213117]: New goal received, planning path...
> [ INFO] [1720197665.004753003]: Path planning successful.
> [ WARN] [1720197665.004944464]: Received empty path
> ```

Oh yes Ill look into the 'vehicle_launch` changes. Would it be possible for you to please check if the required rostopics are being subscribed/published properly? Im aware of `/search_node/trajectory_position` that needs to be published and odom + goal topic that needs to be subscribed.

**Comment by rien88** [comment]
Please merge main into this branch

File: src/planner/plan_manage/src/ego_replan_fsm.cpp [hunk 1]
```diff
@@ -43,14 +44,17 @@ namespace ego_planner
     exec_timer_ = nh.createTimer(ros::Duration(0.01), &EGOReplanFSM::execFSMCallback, this);
     safety_timer_ = nh.createTimer(ros::Duration(0.05), &EGOReplanFSM::checkCollisionCallback, this);
 
-    odom_sub_ = nh.subscribe("odom_world", 1, &EGOReplanFSM::odometryCallback, this);
+    odom_sub_ = nh.subscribe("/mavros/odometry/out", 1, &EGOReplanFSM::odometryCallback, this);
```
Comment (rien88) [inline src/planner/plan_manage/src/ego_replan_fsm.cpp:47]: This topic isn't used, the odometry topic is `/mavros/local_position/odom`. And can you include the message type template like the lines below?

File: src/planner/plan_manage/src/ego_replan_fsm.cpp [hunk 2]
```diff
@@ -43,14 +44,17 @@ namespace ego_planner
...
+    goal_sub = nh.subscribe<geometry_msgs::PoseStamped>("/goal", 10000, &EGOReplanFSM::goalCb, this);
```
Comment (rien88) [inline src/planner/plan_manage/src/ego_replan_fsm.cpp:48]: goal_sub needs trailling _

File: src/planner/plan_manage/src/ego_replan_fsm.cpp [hunk 3]
```diff
@@ -89,6 +94,68 @@ namespace ego_planner
...
+            goals_.back() = new_goal;  // Update the last element to the new goal
+        }
+
+        // Assume current goal is always the last element in the vector
+        Eigen::Vector3d current_goal = goals_.back();
+
+        bool success = planner_manager_->planGlobalTraj(odom_pos_, odom_vel_, Eigen::Vector3d::Zero(), current_goal, Eigen::Vector3d::Zero(), Eigen::Vector3d::Zero());
+        if (success) {
+            ROS_INFO("Path planning successful.");
+            auto traj = planner_manager_->getGlobalTrajectory();  // This method needs to be implemented
```
Comment (rien88) [inline src/planner/plan_manage/src/ego_replan_fsm.cpp:132]: This method has a nonempty implementation, is this comment addressed? Also I'm still getting empty paths even when I locally corrected the odom topic so something in here isn't working

**Comment by rien88** [comment]
Pulled and tested, and ego planner crashed :cry: 
Screenshot from 2024-07-11 08-55-27

**Comment by rien88** [comment]
With the latest changes, I am not getting any paths, or rather, the published paths are empty. On the plus side, I did see the path planner get multiple goals and not crash, so progress! :laughing:

**Comment by rien88** [comment]
Closing because we decided not to integrate this path planner.

**Pull request closed without merging**